### PR TITLE
Bugfix: Fix copy file to s3 source and destination file names being incorrect

### PR DESCRIPTION
**Pull request by naemono**
I think #17 may have created additional bugs.  Note I'm trying to upload a single file to s3 from my local filesystem:

```bash
$ s3kor --verbose cp --acl=public-read --region=us-east-2 /Users/naemono/git/sensu-k8s/scripts/go/cluster-dns-check/bin/cluster-dns-check s3://my-bucket/sensu/assets/
Files            1 / 1 [==============================================================================] Done!
Size 10.3 MB / 10.3 MB [==============================================================================] 100 % 8.3 GB/s Done!
open /Users/naemono/git/sensu-k8s/scripts/go/cluster-dns-check/bin/cluster-dns-check/Users/naemono/git/sensu-k8s/scripts/go/cluster-dns-check/bin/cluster-dns-check: not a directory
```

Check the path it's trying to open, which is not correct.  There also is an issue with the destination in s3, as it (for me) wrote

```
s3://my-bucket/sensu/assets/Users/naemono/git/sensu-k8s/scripts/go/cluster-dns-check/bin/cluster-dns-check 
```

Note that it's prepending my local directory to the name.

This PR fixes the issues *for me*, but I haven't tested all potential use cases.

Also, let me know if you want me to rebase those commits down to a single commit, or if you squash on merge.

### CODE REVIEW

**Comment by sethkor** [comment]
Hey just going through some of the recent things now, please be patient with me

**Comment by sethkor** [comment]
#22 Has created a conflict.  Are you able to update your PR please @naemono ?

**Comment by naemono** [comment]
Sure thing @sethkor .  I'll get this updated today.  No worries on the delays.

File: copy.go [hunk 1]
```diff
@@ -2,20 +2,19 @@ package main
...
 	"time"
 
+	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/s3"
-
-	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/vbauerster/mpb/v5"
+	"github.com/vbauerster/mpb/v5/decor"
```
Comment (naemono) [inline copy.go:17]: best practice is to have stdlib imports at the top, and 3rd party grouped next

File: copy.go [hunk 2]
```diff
@@ -136,15 +135,10 @@ func (cp BucketCopier) optimizeStorageClass(size int64) string {
 	return s3.StorageClassStandard
 }
 
-func (cp *BucketCopier) copyFile(file fileJob, singleFile bool) {
```
Comment (naemono) [inline copy.go:139]: This `singleFile` argument is unnecessary.  I tested all s3 options to verify (recursive, and non-recursive single-file copies)

File: s3kor.go [hunk 3]
```diff
@@ -23,7 +23,7 @@ import (
 var (
 	app = kingpin.New("s3kor", "s3 tools using golang concurency")
 
-	pCustomEndpointUrl = app.Flag("custom-endpoint-url", "AWS S3 Custom Endpoint URL").String()
+	pCustomEndpointURL = app.Flag("custom-endpoint-url", "AWS S3 Custom Endpoint URL").String()
```
Comment (naemono) [inline s3kor.go:26]: golint was complaining about this

File: s3kor.go [hunk 4]
```diff
@@ -151,28 +151,23 @@ func setUpLogger() {
...
-		CredentialsChainVerboseErrors: aws.Bool(true),
-		MaxRetries:                    aws.Int(30),
-		S3ForcePathStyle:              aws.Bool(true),
-	}
+	fmt.Printf("Using custom endpoint [%+v] on region [%+v]\n", *pCustomEndpointURL, *pRegion)
+	config.Endpoint = aws.String(*pCustomEndpointURL)
+	config.S3ForcePathStyle = aws.Bool(true)
 
-	if *pRegion != "" {
+	if pRegion != nil && *pRegion != "" {
```
Comment (naemono) [inline s3kor.go:170]: this is just a simplification of the logic, and some `!= nil` checks to ensure it doesn't panic

**Pull request merged**